src/compiler/transclude.js

```diff
@@ -87,11 +87,13 @@ function transcludeContent (el, raw) {
       select = outlet.getAttribute('select')
       if (select) {  // select content
         selected = raw.querySelectorAll(select)
-        outlet.content = _.toArray(
-          selected.length
-            ? selected
-            : outlet.childNodes
-        )
+        if (selected.length) {
```

Comment:
One possible issue here: if all nodes in `selected` are nested ones (thus should be filtered out), we end up outputting empty content, but in that case we should actually use the fallback content (`outlet.childNodes`).

According to http://w3c.github.io/webcomponents/spec/shadow/#dfn-pool-distribution-algorithm, the algorithm here should be:

- For each direct child nodes of `raw`, check if it matches `select`.
- If yes, push it into `selected`.
- If `selected` ends up empty, use `outlet.childNodes` instead.

This also means we need a `matches()` helper in `util/dom.js`:

``` js
  var p = Element.prototype
	var matches = p.matches || p.webkitMatchesSelector || p.mozMatchesSelector || p.msMatchesSelector
  exports.matches = function (el, selector) {
    return matches.call(el, selector)
  }
```